src/main/java/org/springframework/data/couchbase/repository/query/AbstractN1qlBasedQuery.java

```diff
@@ -71,14 +78,22 @@ protected AbstractN1qlBasedQuery(CouchbaseQueryMethod queryMethod, CouchbaseOper
    */
   protected abstract boolean useGeneratedCountQuery();
 
-  protected abstract Statement getStatement(ParameterAccessor accessor, Object[] runtimeParameters);
+  protected abstract Statement getStatement(ParameterAccessor accessor, Object[] runtimeParameters, ReturnedType returnedType);
 
   protected abstract JsonValue getPlaceholderValues(ParameterAccessor accessor);
 
   @Override
   public Object execute(Object[] parameters) {
     ParametersParameterAccessor accessor = new ParametersParameterAccessor(queryMethod.getParameters(), parameters);
-    Statement statement = getStatement(accessor, parameters);
+
```

Comment:
This looks like we're issuing a query and then basically map the results back onto a DTO. Is there a way we can instruct the query to let the query execution return the requested type in the first place?